packages/imt.sol/contracts/internal/InternalLeanIMT.sol

```diff
@@ -90,7 +90,7 @@ library InternalLeanIMT {
                 revert LeafAlreadyExists();
             }
 
-            self.leaves[leaves[i]] = self.size + i;
```

Comment:
Can you explain why adding '+ 1' is correct here?

I couldn't find a concrete answer, probably I'm missing some implementation details. Following your test below, I expected `indexOf(1) = 1` due to `self.size = 0` and `i = 0`, but it seems to return `indexOf(1) = 0` as you wrote. Hmm, just waiting for your feedback and making this a general comment.

Reply:
Hey! You can find more info in the issue description #125.

The function `indexOf` should work like the JavaScript `indexOf` function. In JavaScript, if we have the array `[1, 2]`, `indexOf(1)` should return 0.

The values saved before were the real positions (`self.leaves[leaves[i]] = self.size + i;`), but the positions that should be saved in `self.leaves` is the real one + 1 (`self.leaves[leaves[i]] = self.size + 1 + i;`). That's why the `indexOf` function in the contracts returns `self.leaves[leaf] - 1`.

Let me know if it is more clear now.

Reply:
Thank you so much for the explanation. Now makes definitely much more sense :blush: